### ISSUE TITLE
regarding my tensorflow library

### ISSUE THREAD

**Issue by 2Arnav0** [spyder-ide/spyder#21853]
### What steps will reproduce the problem?

first of all my kernel downloaded the old version of tensorflow library for my project simulationand now it not even updating the library and it is taking too much time for testing my project simulation.

### Traceback
```python-traceback
Traceback (most recent call last):
  File "A:\PYTHON\Lib\asyncio\tasks.py", line 490, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "A:\PYTHON\Lib\asyncio\futures.py", line 198, in result
    raise exc
  File "A:\PYTHON\Lib\asyncio\tasks.py", line 269, in __step
    result = coro.throw(exc)
             ^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\jupyter_client\manager.py", line 642, in _async_wait
    await asyncio.sleep(pollinterval)
  File "A:\PYTHON\Lib\asyncio\tasks.py", line 639, in sleep
    return await future
           ^^^^^^^^^^^^
  File "A:\PYTHON\Lib\asyncio\futures.py", line 287, in __await__
    yield self  # This tells Task to wait for completion.
    ^^^^^^^^^^
  File "A:\PYTHON\Lib\asyncio\tasks.py", line 339, in __wakeup
    future.result()
  File "A:\PYTHON\Lib\asyncio\futures.py", line 198, in result
    raise exc
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "A:\PYTHON\Lib\site-packages\jupyter_client\manager.py", line 430, in _async_finish_shutdown
    await asyncio.wait_for(
  File "A:\PYTHON\Lib\asyncio\tasks.py", line 492, in wait_for
    raise exceptions.TimeoutError() from exc
TimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "A:\PYTHON\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\client.py", line 772, in _restart_thread_main
    self.shellwidget.kernel_manager.restart_kernel(
  File "A:\PYTHON\Lib\site-packages\jupyter_client\utils.py", line 30, in wrapped
    raise e
  File "A:\PYTHON\Lib\site-packages\jupyter_client\utils.py", line 27, in wrapped
    return loop.run_until_complete(future)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\nest_asyncio.py", line 90, in run_until_complete
    return f.result()
           ^^^^^^^^^^
  File "A:\PYTHON\Lib\asyncio\futures.py", line 203, in result
    raise self._exception.with_traceback(self._exception_tb)
  File "A:\PYTHON\Lib\asyncio\tasks.py", line 267, in __step
    result = coro.send(None)
             ^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\jupyter_client\manager.py", line 541, in _async_restart_kernel
    await ensure_async(self.shutdown_kernel(now=now, restart=True))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\jupyter_client\utils.py", line 30, in wrapped
    raise e
  File "A:\PYTHON\Lib\site-packages\jupyter_client\utils.py", line 27, in wrapped
    return loop.run_until_complete(future)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\nest_asyncio.py", line 90, in run_until_complete
    return f.result()
           ^^^^^^^^^^
  File "A:\PYTHON\Lib\asyncio\futures.py", line 203, in result
    raise self._exception.with_traceback(self._exception_tb)
  File "A:\PYTHON\Lib\asyncio\tasks.py", line 267, in __step
    result = coro.send(None)
             ^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\jupyter_client\manager.py", line 90, in wrapper
    raise e
  File "A:\PYTHON\Lib\site-packages\jupyter_client\manager.py", line 82, in wrapper
    out = await method(self, *args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\jupyter_client\manager.py", line 504, in _async_shutdown_kernel
    await ensure_async(self.finish_shutdown(restart=restart))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\jupyter_client\utils.py", line 30, in wrapped
    raise e
  File "A:\PYTHON\Lib\site-packages\jupyter_client\utils.py", line 27, in wrapped
    return loop.run_until_complete(future)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\nest_asyncio.py", line 90, in run_until_complete
    return f.result()
           ^^^^^^^^^^
  File "A:\PYTHON\Lib\asyncio\futures.py", line 203, in result
    raise self._exception.with_traceback(self._exception_tb)
  File "A:\PYTHON\Lib\asyncio\tasks.py", line 267, in __step
    result = coro.send(None)
             ^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\jupyter_client\manager.py", line 436, in _async_finish_shutdown
    await ensure_async(self._send_kernel_sigterm())
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\jupyter_client\utils.py", line 30, in wrapped
    raise e
  File "A:\PYTHON\Lib\site-packages\jupyter_client\utils.py", line 27, in wrapped
    return loop.run_until_complete(future)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\nest_asyncio.py", line 90, in run_until_complete
    return f.result()
           ^^^^^^^^^^
  File "A:\PYTHON\Lib\asyncio\futures.py", line 203, in result
    raise self._exception.with_traceback(self._exception_tb)
  File "A:\PYTHON\Lib\asyncio\tasks.py", line 267, in __step
    result = coro.send(None)
             ^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\spyder\plugins\ipythonconsole\utils\manager.py", line 118, in _async_send_kernel_sigterm
    await self.kill_proc_tree(self.provisioner.process.pid)
  File "A:\PYTHON\Lib\site-packages\spyder\plugins\ipythonconsole\utils\manager.py", line 57, in kill_proc_tree
    children = parent.children(recursive=True)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "A:\PYTHON\Lib\site-packages\psutil\__init__.py", line 277, in wrapper
    raise NoSuchProcess(self.pid, self._name, msg=msg)
psutil.NoSuchProcess: process no longer exists (pid=10320)
```

## Versions

* Spyder version: 5.4.3  (conda)
* Python version: 3.11.5 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                             :  1.4.0 (OK)
chardet >=2.0.0                                  :  4.0.0 (OK)
cloudpickle >=0.5.0                              :  2.2.1 (OK)
cookiecutter >=1.6.0                             :  1.7.3 (OK)
diff_match_patch >=20181111                      :  20200713 (OK)
intervaltree >=3.0.2                             :  3.1.0 (OK)
IPython >=7.31.1,<9.0.0,!=8.8.0,!=8.9.0,!=8.10.0 :  8.15.0 (OK)
jedi >=0.17.2,<0.19.0                            :  0.18.1 (OK)
jellyfish >=0.7                                  :  1.0.1 (OK)
jsonschema >=3.2.0                               :  4.17.3 (OK)
keyring >=17.0.0                                 :  23.13.1 (OK)
nbconvert >=4.0                                  :  6.5.4 (OK)
numpydoc >=0.6.0                                 :  1.5.0 (OK)
paramiko >=2.4.0                                 :  2.8.1 (OK)
parso >=0.7.0,<0.9.0                             :  0.8.3 (OK)
pexpect >=4.4.0                                  :  4.8.0 (OK)
pickleshare >=0.4                                :  0.7.5 (OK)
psutil >=5.3                                     :  5.9.0 (OK)
pygments >=2.0                                   :  2.15.1 (OK)
pylint >=2.5.0,<3.0                              :  2.16.2 (OK)
pylint_venv >=2.1.1                              :  2.3.0 (OK)
pyls_spyder >=0.4.0                              :  0.4.0 (OK)
pylsp >=1.7.2,<1.8.0                             :  1.7.2 (OK)
pylsp_black >=1.2.0                              :  1.2.1 (OK)
qdarkstyle >=3.0.2,<3.2.0                        :  3.0.2 (OK)
qstylizer >=0.2.2                                :  0.2.2 (OK)
qtawesome >=1.2.1                                :  1.2.2 (OK)
qtconsole >=5.4.2,<5.5.0                         :  5.4.2 (OK)
qtpy >=2.1.0                                     :  2.2.0 (OK)
rtree >=0.9.7                                    :  1.0.1 (OK)
setuptools >=49.6.0                              :  68.0.0 (OK)
sphinx >=0.6.6                                   :  5.0.2 (OK)
spyder_kernels >=2.4.3,<2.5.0                    :  2.4.4 (OK)
textdistance >=4.2.0                             :  4.2.1 (OK)
three_merge >=0.1.1                              :  0.1.1 (OK)
watchdog >=0.10.3                                :  2.1.6 (OK)
zmq >=22.1.0                                     :  23.2.0 (OK)

# Optional:
cython >=0.21                                    :  None (NOK)
matplotlib >=3.0.0                               :  3.7.2 (OK)
numpy >=1.7                                      :  1.24.3 (OK)
pandas >=1.1.1                                   :  2.0.3 (OK)
scipy >=0.17.0                                   :  1.11.1 (OK)
sympy >=0.7.3                                    :  1.11.1 (OK)
```

**Comment by ccordoba12** [comment]
Hey @2Arnav0, thanks for reporting. I think this problem is fixed in our latest version available (5.5.1). Please update by closing Spyder, opening a terminal and running there

    pip install -U spyder

since it seems you used pip to install it before.

Let us know if that works for you.

**Comment by ccordoba12** [comment]
Closing due to lack of response.
